### PR TITLE
feat: Expose routeName in navigate/replace with transition methods in stacked_services

### PR DESCRIPTION
**Pull request by iamelio**
This PR closes issue #685

### CODE REVIEW

File: packages/stacked_themes/lib/src/theme_manager.dart [hunk 1]
```diff
@@ -196,10 +196,17 @@ You can supply either a list of ThemeData objects to the themes property or a li
...
   /// Swaps between the light and dark ThemeMode
-  void toggleDarkLightTheme() {
+  ///
+  /// If [disablePersistence] is set to true, the selected theme will
+  /// not persist across app restarts.
+  void toggleDarkLightTheme([bool disablePersistence = false]) {
     _selectedThemeMode =
         _selectedThemeMode == ThemeMode.dark ? ThemeMode.light : ThemeMode.dark;
 
+    if (disablePersistence == false) {
```
Comment (Eimen2018) [inline packages/stacked_themes/lib/src/theme_manager.dart]: Should this be added to this PR?

Comment (iamelio) [reply packages/stacked_themes/lib/src/theme_manager.dart]: Yes please, if that would be easier to review, so you can close the other PR and the relevant issue(s).

Comment (Eimen2018) [reply packages/stacked_themes/lib/src/theme_manager.dart]: I just find it out of scope. doesn't relate to the PR title in any way.

Comment (iamelio) [reply packages/stacked_themes/lib/src/theme_manager.dart]: Alright, I understand, you can remove it, thanks.

Comment (Eimen2018) [reply packages/stacked_themes/lib/src/theme_manager.dart]: What do you want me to remove the whole PR?

Comment (iamelio) [reply packages/stacked_themes/lib/src/theme_manager.dart]: I was thinking to drop that particular commit actually, so you would only deal with the code relevant to this PR

**Comment by iamelio** [comment]
Hey @Eimen2018, I've cleaned up the PR to contain only the code to fix issue #685 and synced up with the latest on the `master` branch.

**Comment by Eimen2018** [comment]
Awesome! all that's left now is to bump the version and add a change log @thecokerdavid

**Comment by iamelio** [comment]
Hi @Eimen2018, it's done! :)

**Review by Eimen2018 — APPROVED** [review]
LGTM

**Comment by FilledStacks** [comment]
Doooooope Thanks for reviewing @Eimen2018

**Pull request merged**